diff --git a/core.js b/core.js
index d653e66a4056c27cca777d4e25222acae3b2ec85..a91741d67df85fd9627889a6c7197ac4e6a3a813 100644
--- a/core.js
+++ b/core.js
@@ -1415,8 +1415,7 @@ async function _fromTokenizer(tokenizer) {
 }

 const stream = readableStream => new Promise((resolve, reject) => {
-	// Using `eval` to work around issues when bundling with Webpack
-	const stream = eval('require')('stream'); // eslint-disable-line no-eval
+	const stream = require('node:stream');

 	readableStream.on('error', reject);
 	readableStream.once('readable', async () => {
