diff --git a/umd/vudio.js b/umd/vudio.js
index d0d1127e57125ad4e77442af2db4a26998c7b385..c0b66bd4327c65c31dc6e588bfa4ae6ec70bd3b8 100644
--- a/umd/vudio.js
+++ b/umd/vudio.js
@@ -147,7 +147,6 @@
 
             source.connect(this.analyser);
             this.analyser.fftSize = this.option.accuracy * 2;
-            this.analyser.connect(audioContext.destination);
 
             this.freqByteData = new Uint8Array(this.analyser.frequencyBinCount);
 
@@ -207,7 +206,6 @@
 
             source.connect(this.analyser);
             this.analyser.fftSize = this.option.accuracy * 2;
-            this.analyser.connect(audioContext.destination);
         },
 
         __rebuildData : function (freqByteData, horizontalAlign) {
